### PR TITLE
Upgrade to protobuf-3.0.0-beta-1

### PR DESCRIPTION
**Pull request by zhangkun83**
protobuf-nano, however, is alpha-4.

### CODE REVIEW

**Comment by zhangkun83** [comment]
The nano generator from protobuf starts to add `.nano` to the message package names, while protobuf doesn't provide a helper function to get the proper class name for now. We have to use a hack to add `nano` for now. Protobuf team will add the helper function in the next release. (Tracking issue #900).

Protobuf also provide an in-file option to go back to the original package name. Because it is carried in the `FileOptions` which is a proto message, checking this option requires us to generate this C++ class thus would require some change to the build of `grpc-compiler`. Since protobuf team plans to eliminate this option for good, I don't think it's worth the effort to support it. @ejona86 WDYT?

**Comment by zhangkun83** [comment]
Never mind. It's already generated and available in ``descriptor.pb.h``.

File: README.md [hunk 1]
```diff
@@ -69,7 +69,7 @@ For protobuf-based codegen integrated with the Maven build system, you can use
           protobuf-java directly, you will be transitively depending on the
           protobuf-java version that grpc depends on.
         -->
-        <protocArtifact>com.google.protobuf:protoc:3.0.0-alpha-3.1:exe:${os.detected.classifier}</protocArtifact>
+        <protocArtifact>com.google.protobuf:protoc:3.0.0-beta-1:exe:${os.detected.classifier}</protocArtifact>
```
Comment (ejona86) [inline README.md]: We don't want to change these yet, since these are based on the last release.

Comment (ejona86) [inline README.md]: Or is it safe to do now? What was the problem we had earlier?

**Comment by ejona86** [comment]
@zhangkun83, other than the changes to README.md, LGTM

File: README.md (hunk 1 above)
Comment (zhangkun83) [inline README.md]: Agreed. We don't change it now.

Right it's not safe to do so. The problem we had earlier was that the dependency on grpc introduced a transitive dependency on a certain version of protobuf runtime, which was different from the version of `protocArtifact`.

**Pull request merged**